lib/screens/player_screen.dart

```diff
@@ -173,3 +163,107 @@ class _PlayerScreenContent extends StatelessWidget {
     );
   }
 }
+
+enum PlayerHideable {
+  bigPlayButton(14, 14),
+  queueButton(0, 27),
+  progressSlider(0, 14),
+  twoLineTitle(0, 27),
+  codecInfo(0, 20),
+  loopShuffleButtons(96, 0);
+
+  final double width;
+  final double height;
+
+  const PlayerHideable(this.width, this.height);
+}
+
+class PlayerHideableController {
+  final verticalHideOrder = [
+    PlayerHideable.bigPlayButton,
+    PlayerHideable.queueButton,
+    PlayerHideable.codecInfo,
+    PlayerHideable.twoLineTitle,
+    PlayerHideable.progressSlider
+  ];
+
+  List<PlayerHideable> _visible = [];
+  Size? _target;
+
+  void setSizePortrait(Size size) {
+    _visible = List.from(PlayerHideable.values);
+
+    var maxPadding =
+        FinampSettingsHelper.finampSettings.playerScreenCoverMinimumPadding + 8;
+    var targetHeight =
+        size.height + size.width * ((maxPadding / 100.0) * 2 - 1);
+    var targetWidth = size.width;
+    _target = Size(targetWidth, targetHeight);
+
+    if (_getSize().width >= targetWidth) {
+      _visible.remove(PlayerHideable.bigPlayButton);
+      if (_getSize().width >= targetWidth) {
+        _visible.add(PlayerHideable.bigPlayButton);
```

Comment:
The side effect of changing `_visible` to affect the result of `_getSize()` isn't super obvious, and it's really hard to tell what is going on. Is there a more explicit way to configure this, that also scales well in the future?

Reply:
I've refactored a couple things and added a bunch of comments to hopefully make the everything a bit clearer.  Tell me if it actually helps.